hibernate-search-plugins/org.jboss.tools.hibernate.search.runtime.common/src/org/jboss/tools/hibernate/search/runtime/common/AbstractFacadeFactory.java

```diff
@@ -0,0 +1,85 @@
+package org.jboss.tools.hibernate.search.runtime.common;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import org.jboss.tools.hibernate.runtime.common.IFacade;
+import org.jboss.tools.hibernate.runtime.common.Util;
+import org.jboss.tools.hibernate.runtime.spi.ISession;
+import org.jboss.tools.hibernate.runtime.spi.ISessionFactory;
+import org.jboss.tools.hibernate.search.runtime.spi.IAnalyzer;
+import org.jboss.tools.hibernate.search.runtime.spi.IFullTextSession;
+import org.jboss.tools.hibernate.search.runtime.spi.IQueryParser;
+import org.jboss.tools.hibernate.search.runtime.spi.ISearchFactory;
+
+public abstract class AbstractFacadeFactory implements IFacadeFactory {
```

Comment:
The design is the same implemented for hibernate plugin: 
https://github.com/jbosstools/jbosstools-hibernate/blob/master/plugins/org.jboss.tools.hibernate.runtime.common/src/org/jboss/tools/hibernate/runtime/common/AbstractFacadeFactory.java

The architecture is described here by @koentsje http://tools.jboss.org/blog/hibernate-tools-changes.html